### ISSUE TITLE
Support epoch timestamp for date filter

### ISSUE THREAD

**Issue by pirsquare** [jg-rp/liquid#67]
Currently date filter won't work for epoch timestamp. In other libraries, epoch timestamp works.
https://liquidjs.com/playground.html

```
{% assign created_timestamp = "1645173284" -%}
{{ created_timestamp | date: "%d %b %Y" }}
```

Will render
```
18 Feb 2022
```

We can add a minor update to date filter to support epoch timestamp
```
try:
    dat = datetime.datetime.fromtimestamp(int(dat), datetime.timezone.utc)
except:
    pass
```

Full function update
```
@with_environment
@liquid_filter
@functools.lru_cache(maxsize=10)
def date(
    dat: Union[datetime.datetime, str], fmt: str, *, environment: Environment
) -> str:
    """Formats a datetime according the the given format string."""
    if is_undefined(dat):
        return ""

    if is_undefined(fmt):
        return str(dat)

    # try if it's epoch timestamp
    try:
        dat = datetime.datetime.fromtimestamp(int(dat), datetime.timezone.utc)
    except:
        pass

    if isinstance(dat, str):
        if dat in ("now", "today"):
            dat = datetime.datetime.now()
        else:
            dat = parser.parse(dat)

    if not isinstance(dat, (datetime.datetime, datetime.date)):
        raise FilterArgumentError(
            f"date expected datetime.datetime, found {type(dat).__name__}"
        )

    rv = dat.strftime(fmt)

    if environment.autoescape and isinstance(fmt, Markup):
        return Markup(rv)
    return 
```

**Comment by jg-rp** [comment]
Hi @pirsquare,

Thank you. The `date` filter in [the reference implementation](https://github.com/Shopify/liquid) of Liquid supports timestamps too.

This will be fixed in the next release. Until then, you can register the new `date` implementation, or an implementation of your own, with an `Environment` like this.

```python
import datetime
from typing import Union
from dateutil import parser

from liquid import Environment
from liquid import Markup
from liquid import is_undefined

from liquid.filter import liquid_filter
from liquid.filter import with_environment

from liquid.exceptions import FilterArgumentError


@with_environment
@liquid_filter
def date(
    dat: Union[datetime.datetime, str, int],
    fmt: str,
    *,
    environment: Environment,
) -> str:
    """Formats a datetime according the the given format string."""
    if is_undefined(dat):
        return ""

    if is_undefined(fmt):
        return str(dat)

    if isinstance(dat, str):
        if dat in ("now", "today"):
            dat = datetime.datetime.now()
        elif dat.isdigit():
            # The reference implementation does not support string
            # representations of negative integers either.
            dat = datetime.datetime.fromtimestamp(int(dat))
        else:
            try:
                dat = parser.parse(dat)
            except parser.ParserError:
                # Input is returned unchanged. This is consistent
                # with the reference implementation.
                return str(dat)
    elif isinstance(dat, int):
        try:
            dat = datetime.datetime.fromtimestamp(dat)
        except (OverflowError, OSError):
            # Testing on Windows shows that it can't handle some
            # negative integers.
            return str(dat)

    if not isinstance(dat, (datetime.datetime, datetime.date)):
        raise FilterArgumentError(
            f"date expected datetime.datetime, found {type(dat).__name__}"
        )

    rv = dat.strftime(fmt)

    if environment.autoescape and isinstance(fmt, Markup):
        return Markup(rv)
    return rv


# Example of replacing the built-in `date` filter
env = Environment()
env.add_filter("date", date)

t = env.from_string("{{ '1152098955' | date: '%m/%d/%Y' }}")
print(t.render())
```

**Comment by pirsquare** [comment]
Yes I actually did that, thankfully you had a section on it to replace existing filter. Great library and thank you for your work!

**Comment by jg-rp** [comment]
Fixed in version [1.4.3](https://pypi.org/project/python-liquid/), now released.
